### PR TITLE
add hw9

### PR DESCRIPTION
**Pull request by adlerberg0**

### CODE REVIEW

File: hw9/task1.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+4
+6
+#>>> list(merge_sorted_files(["file1.txt", "file2.txt"]))
+#[1, 2, 3, 4, 5, 6]
+"""
+from pathlib import Path
+from typing import Iterator, List, Union
+
+
+def merge_helping_func(data_list: list) -> list:
```
Comment (AlekseiAfanasev) [inline hw9/task1.py]: this can be much simpler. try to use `sorted()`

File: hw9/task2.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+Do it both ways: as a class and as a generator.
+#>>> with supressor(IndexError):
+#...    [][2]
+"""
+from contextlib import contextmanager
+from types import TracebackType
+from typing import Optional, Type  # NOQA
+
+# how to fix this?
+# hw9/task2.py:9:1: TYP001 guard import by `if False:  # TYPE_CHECKING`: Type (not in 3.5.0, 3.5.1)
```
Comment (AlekseiAfanasev) [inline hw9/task2.py]: you can try 
```
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from typing import TYPE
```

Comment (adlerberg0) [reply hw9/task2.py]: Done

File: hw9/task1.py (hunk 1 above)
Comment (adlerberg0) [reply hw9/task1.py]: Done

**Review by AlekseiAfanasev — APPROVED** [review]

**Pull request merged**